osquery/tables/system/windows/bam.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/system/time.h>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+constexpr auto kBamRegPath =
+    "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Services\\bam\\%%\\%%";
+
+// Get last exeution time
+auto last_execute_time(std::string& assist_data) {
```

Comment:
As was mentioned in #5539 let's abstract this into a generic windows header file somewhere and have it used across both. Perhaps the declaration could live in the `osquery/tables/system/windows/userassist.h`?

Reply:
I added it to userassist.h
once #5539 is merged i can rebase this PR and change it to call the function from userassist.h